ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/vlan_interfaces.py

```diff
@@ -46,6 +46,17 @@ def vlan_interfaces(self) -> dict | None:
         return None
 
     def _get_vlan_interface_config_for_svi(self, svi, vrf, tenant) -> dict:
+        def _check_virtual_router_mac_address(vlan_interface_config: dict, variables: list):
+            """
+            Check if any variable in the list of variables is not None in vlan_interface_config
+            and if it is the case, raise an Exception if virtual_router_mac_address is None
+            """
+            if any(vlan_interface_config[var] for var in variables) and self._virtual_router_mac_address is None:
+                quoted_vars = [f"'{var}'" for var in variables]
```

Comment:
!!! Optional

This won't create any intermediate list.
```suggestion
                quoted_vars = map(lambda var: f"'{var}'", variables)
```

Reply:
Not an optimization for smaller lists but definitely good for bigger lists.

Reply:
Thank you for the suggestion.
I don't think it would be better in this case. It would just make it harder to read.
In general `map`+`lambda` will probably be slower that `iter(f"{var}" for var in variables)`.